_posts/2019-08-26-how-to-use-gruntwork-service-catalog.adoc

```diff
@@ -0,0 +1,1657 @@
+  https://blog.gruntwork.io/tagged/gruntwork-newsletter[Gruntwork Newsletter] to get better infrastructure through a
+  version number bump.
+
+Get commercial support::
+  Get access to a team of DevOps experts who can help you set up your infrastructure, design highly available and
+  scalable systems, automate your builds and deployments, troubleshoot issues, and avoid gotchas and pitfalls. Reach out
+  to us via Slack, email, and phone/video calls, get code reviews, prioritized bug fixes, and SLAs on response times.
+
+=== What you'll learn in this guide
+
+This guide consists of three main sections:
+
+<<core_concepts>>::
+  An overview of the core concepts you need to understand to use the Gruntwork Service Catalog, including a look into
+  how the Service Catalog is designed, how the Reference Architecture is designed, how we build production-grade
+  infrastructure, and how to make use of infrastructure as code, Terraform, Terragrunt, Packer, Docker, immutable
+  infrastructure, versioning, automated testing, and more.
+
+<<how_to_use_the_catalog>>::
+  An overview of how to use the Gruntwork Service Catalog.
+
+<<next_steps>>::
+  What to do next once you've finished reading this guide.
+
+Feel free to read the guide from start to finish or skip around to whatever part interests you!
+
+[[core_concepts]]
+== Core concepts
+
+=== The Gruntwork Service Catalog
+
+// TODO: add a screenshot of the service catalog
+
+The Gruntwork Service Catalog is a collection of over 300,000 lines of reusable, battle-tested infrastructure code.
+that is organized into 40+ GitHub repos, some public and open source, and some private and only accessible to Gruntwork
+customers. Each repo is focused on one type of infrastructure: e.g., there is one repo that contains code for deploying
+and managing Kubernetes on AWS, one repo with code for deploying and managing the ELK stack, one repo that
+contains a collection of CI / CD code, and so on.
+
+The code in the Gruntwork Service Catalog is written using a combination of:
+
+Terraform::
+  Used to to define and manage most of the basic infrastructure, such as servers, databases, load balancers, and
+  networking. More on <<terraform>> later.
+
+Bash::
+  Used for _install scripts_ (e.g., a Bash script that you can use to create Linux VM image with Elasticsearch
+  installed) and _run scripts_ (e.g., you use Terraform to deploy a cluster of server with the Elasticsearch VM image
+  on it and configure the servers to execute a Bash script during boot that can auto-discover and bootstrap the
+  Elasticsearch cluster).
+
+Python::
+  Used for more complicated scripts, especially those that need to run on other operating systems (e.g., Windows)
+  and/or those that need to be called directly from Terraform (e.g., to fill in some missing functionality).
+
+Go::
+  Used to build complex, cross-platform CLI applications. E.g., We have an app called `ssh-grunt` you can run on each
```

Comment:
and tests, using this super mega awesome lib, also developed by us, click here

Reply:
Oh yea! Updated: https://github.com/gruntwork-io/gruntwork-io.github.io/pull/151/commits/4603b48a336569db90352172790bc29347291c75